### PR TITLE
[BH-1657][BH-1833][BH-1854] Add WFI and SDRAM self-refresh mode

### PR DESCRIPTION
**Pull request by Lefucjusz**
* Added mechanism enabling CPU to enter WFI mode 
when the OS is in idle, what results in large power 
consumption reduction.
* Added mechanism to switch SDRAM to self-refresh 
mode before entering WFI, what resulted in further 
power consumption reduction.

Coauthored by @Maciej-Mudita and @dawidwojtas-mudita.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

File: module-bsp/board/rt1051/bellpx/bsp/lpm/WfiController.cpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+namespace bsp
+{
+    namespace
+    {
+        /* RTC wakes up CPU every minute, so go to sleep only if next timer will
+         * trigger after more than minute - this way no event will ever be missed */
+        constexpr auto timersInactivityTimeMs{60 * utils::time::milisecondsInSecond};
+
+        bool wfiModeAllowed = false;
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply module-bsp/board/rt1051/bellpx/bsp/lpm/WfiController.cpp:22]: This variable is already placed in anonymous namespace, which is effectively the same as `static`

File: module-bsp/board/rt1051/common/startup_mimxrt1052.cpp [hunk 2]
```diff
@@ -716,6 +716,18 @@ __attribute__((section(".after_vectors.reset"))) void ResetISR(void)
         }
     }
 
+    // Load address of OCRAM text section entry in GST
+    SectionTableAddr = &__ocramtext_section_table;
+    // Copy code from SDRAM to OCRAM
+    while (SectionTableAddr < &__ocramtext_section_table_end) {
+        LoadAddr   = *SectionTableAddr++;
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply module-bsp/board/rt1051/common/startup_mimxrt1052.cpp:723]: You're totally right and I'd probably implement this completely differently if I was writing this from the scratch, but this is just copied from a few lines above, which is the example startup code provided by NXP. To keep everything consistent and avoid unnecessary changes in critical startup routines I left it as it was :wink:

File: module-sys/SystemManager/PowerManager.cpp [hunk 3]
```diff
@@ -75,6 +78,7 @@ namespace sys
         cpuFrequencyMonitor.push_back(CpuFrequencyMonitor(lowestLevelName));
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply module-sys/SystemManager/PowerManager.cpp]: Fixed

**Review by Maciej-Mudita — APPROVED** [review]

File: module-bsp/board/rt1051/os/fsl_runtimestat_gpt.c [hunk 4]
```diff
@@ -1,8 +1,12 @@
-// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// Copyright (c) 2017-2023, Mudita Sp. z.o.o. All rights reserved.
 // For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
+#include <fsl_runtimestat_gpt.h>
 #include "fsl_gpt.h"
 
+static const uint32_t milisecondsInSecond = 1000U;
```
Comment (dawidwojtas-mudita) [inline module-bsp/board/rt1051/os/fsl_runtimestat_gpt.c:7]: Typo `millisecondsInSecond` ;)

**Review by dawidwojtas-mudita — APPROVED** [review]

File: module-bsp/board/rt1051/os/fsl_runtimestat_gpt.c (hunk 4 above)
Comment (Lefucjusz) [reply module-bsp/board/rt1051/os/fsl_runtimestat_gpt.c]: Fixed!

**Pull request merged**